### PR TITLE
Added view to add and remove users from the list of speakers.

### PR DESCRIPTION
**Pull request by normanjaeckel**

### CODE REVIEW

**Comment by emanuelschuetze** [comment]
ok

**Comment by normanjaeckel** [comment]
Ready to review.

**Comment by emanuelschuetze** [comment]
ok

File: openslides/agenda/views.py [hunk 1]
```diff
@@ -208,6 +216,137 @@ def get_queryset(self):
...
+            # Retrieve user_id
+            user_id = request.data.get('user')
+
+            # Check permissions and other conditions. Get user instance.
+            if user_id is None:
+                # Add oneself
+                if not self.request.user.has_perm('agenda.can_be_speaker'):
+                    self.permission_denied(request)
+                if item.speaker_list_closed:
+                    raise ValidationError({'detail': _('The list of speakers is closed.')})
```
Comment (ostcar) [inline openslides/agenda/views.py]: Is this exception caught?

File: openslides/agenda/views.py [hunk 2]
```diff
@@ -208,6 +216,137 @@ def get_queryset(self):
...
+            # Check permissions and other conditions. Get speaker instance.
+            if speaker_id is None:
+                # Remove oneself
+                queryset = Speaker.objects.filter(
+                    item=item, user=self.request.user).exclude(weight=None)
+                try:
+                    # We assume that there aren't multiple entries because this
+                    # is forbidden by the Manager's add method. We assume that
+                    # there is only one speaker instance or none.
+                    speaker = queryset.get()
```
Comment (ostcar) [inline openslides/agenda/views.py]: Where is the advantage of this? There is no harm in calling queryset.delete()

File: tests/integration/agenda/test_viewsets.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+
+class ManageSpeaker(TestCase):
+    """
+    Tests managing speakers.
+    """
+    def setUp(self):
+        self.client = APIClient()
+        self.client.login(username='admin', password='admin')
+        self.item = Item.objects.create(title='test_title_aZaedij4gohn5eeQu8fe')
+        self.user = get_user_model().objects.create_user(
```
Comment (ostcar) [inline tests/integration/agenda/test_viewsets.py]: You can use this line, but I understand our integration test, that they test the code with out User app. And that there is at least one user with the pk 1 in it (the admin user). So I will continue to import the User app directly.

**Comment by ostcar** [comment]
You can merge this if you like

File: openslides/agenda/views.py (hunk 1 above)
Comment (normanjaeckel) [inline openslides/agenda/views.py]: Yes, see also http://www.django-rest-framework.org/api-guide/exceptions/#validationerror.

File: openslides/agenda/views.py (hunk 2 above)
Comment (normanjaeckel) [inline openslides/agenda/views.py]: Line 294 is use for both possible results of the line 273 if-clause.

File: tests/integration/agenda/test_viewsets.py (hunk 3 above)
Comment (normanjaeckel) [inline tests/integration/agenda/test_viewsets.py]: I do not understand this remark, neither why we should bind our authentication app (called users at the moment) together with the other app. This binding is not necessary.

File: openslides/agenda/views.py (hunk 1 above)
Comment (ostcar) [inline openslides/agenda/views.py]: ok

File: openslides/agenda/views.py (hunk 2 above)
Comment (ostcar) [inline openslides/agenda/views.py]: I know, but line 286 could also use filter instead of get

Comment (normanjaeckel) [inline openslides/agenda/views.py]: Then I can't catch the Speaker.DoesNotExist exception to validate whether the user is on the list of speakers or a speaker with the given id exists.

**Pull request merged**